src/NzbDrone.Mono.Test/DiskProviderTests/DiskProviderFixture.cs

```diff
@@ -151,11 +151,73 @@ public void should_copy_folder_permissions()
             Syscall.stat(dst, out var dstStat);
             dstStat.st_mode.Should().Be(origStat.st_mode);
 
-            Subject.CopyPermissions(src, dst, false);
+            Subject.CopyPermissions(src, dst);
 
             // Verify CopyPermissions
             Syscall.stat(dst, out dstStat);
             dstStat.st_mode.Should().Be(srcStat.st_mode);
         }
+
+        [Test]
+        public void should_set_file_permissions()
+        {
+            var tempFile = GetTempFilePath();
+
+            File.WriteAllText(tempFile, "File1");
+            SetWritePermissions(tempFile, false);
+
+            // Verify test setup
+            Syscall.stat(tempFile, out var fileStat);
+            NativeConvert.ToOctalPermissionString(fileStat.st_mode).Should().Be("0444");
+
+            Subject.SetPermissions(tempFile, "644");
+            Syscall.stat(tempFile, out fileStat);
+            NativeConvert.ToOctalPermissionString(fileStat.st_mode).Should().Be("0644");
+
+            Subject.SetPermissions(tempFile, "0644");
+            Syscall.stat(tempFile, out fileStat);
+            NativeConvert.ToOctalPermissionString(fileStat.st_mode).Should().Be("0644");
+
+            Subject.SetPermissions(tempFile, "1664");
+            Syscall.stat(tempFile, out fileStat);
+            NativeConvert.ToOctalPermissionString(fileStat.st_mode).Should().Be("1664");
+        }
+
+        [Test]
+        public void should_set_folder_permissions()
+        {
+            var tempPath = GetTempFilePath();
+
+            Directory.CreateDirectory(tempPath);
+            SetWritePermissions(tempPath, false);
+
+            // Verify test setup
+            Syscall.stat(tempPath, out var fileStat);
+            NativeConvert.ToOctalPermissionString(fileStat.st_mode).Should().Be("0555");
+
+            Subject.SetPermissions(tempPath, "644");
+            Syscall.stat(tempPath, out fileStat);
+            NativeConvert.ToOctalPermissionString(fileStat.st_mode).Should().Be("0755");
+
+            Subject.SetPermissions(tempPath, "0644");
+            Syscall.stat(tempPath, out fileStat);
+            NativeConvert.ToOctalPermissionString(fileStat.st_mode).Should().Be("0755");
+
+            Subject.SetPermissions(tempPath, "1664");
+            Syscall.stat(tempPath, out fileStat);
+            NativeConvert.ToOctalPermissionString(fileStat.st_mode).Should().Be("0775");
+
+            Subject.SetPermissions(tempPath, "775");
+            Syscall.stat(tempPath, out fileStat);
+            NativeConvert.ToOctalPermissionString(fileStat.st_mode).Should().Be("0775");
+
+            Subject.SetPermissions(tempPath, "640");
+            Syscall.stat(tempPath, out fileStat);
+            NativeConvert.ToOctalPermissionString(fileStat.st_mode).Should().Be("0750");
+
+            Subject.SetPermissions(tempPath, "0041");
```

Comment:
Doesn't this remove owner permissions? Might need a bit of code to cleanup the test folder afterwards.

Reply:
Yes, but it should be fine because it's a temp file and the file can be removed because the parent folder should have enough permissions.
I added that test, although it's unrealistic, to make sure the function works. I'll remove the file afterwards.

Reply:
Paths retrieved via GetTempFilePath are automatically removed after the test. So if the permissions don't prevent removal then no further changes are needed.